src/cdk/testing/testbed/testbed-harness-environment.ts

```diff
@@ -72,4 +89,53 @@ export class TestbedHarnessEnvironment extends HarnessEnvironment<Element> {
     await this.forceStabilize();
     return Array.from(this.rawRootElement.querySelectorAll(selector));
   }
+
+  /**
+   * Sets up the custom task state ZoneJS interceptor.
+   * @returns an observable that emits whenever the task state changes.
+   */
+  private _setupZoneTaskStateInterceptor(): Observable<TaskState> {
+    if (Zone === undefined) {
+      throw Error('Could not find ZoneJS. For test harnesses running in TestBed, ' +
+          'ZoneJS needs to be installed.');
+    }
+
+    // tslint:disable-next-line:variable-name
+    const ProxyZoneSpec = (Zone as any)['ProxyZoneSpec'] as ProxyZoneType|undefined;
+
+    // If there is no "ProxyZoneSpec" installed, we throw an error and recommend
+    // setting up the proxy zone by pulling in the testing bundle.
+    if (!ProxyZoneSpec) {
+      throw Error(
+        'ProxyZoneSpec is needed for the test harnesses but could not be found. ' +
+        'Please make sure that your environment includes zone.js/dist/zone-testing.js');
+    }
+
+    // Ensure that there is a proxy zone instance set up.
+    ProxyZoneSpec.assertPresent();
+
+    // Get the instance of the proxy zone spec.
+    const zoneSpec = ProxyZoneSpec.get();
+    const currentDelegate = zoneSpec.getDelegate();
+
+    // If there already is a delegate registered in the proxy zone, and it
+    // is type of the custom task state interceptor, we just use that state
+    // observable. This allows us to only intercept Zone once per test
+    // (similar to how `fakeAsync` or `async` work).
+    if (currentDelegate && currentDelegate instanceof TaskStateZoneInterceptor) {
+      return currentDelegate.state;
+    }
+
+    // Since we intercept on environment creation and the fixture has been
+    // created before, we might have missed tasks scheduled before. Fortunately
+    // the proxy zone keeps track of the previous task state, so we can just pass
+    // this as initial state to the task zone interceptor.
+    const newZone = new TaskStateZoneInterceptor(zoneSpec.lastTaskState);
+
+    // Set the new delegate. This means that the "ProxyZone" will delegate
+    // all Zone information to the custom task zone interceptor.
+    zoneSpec.setDelegate(newZone);
```

Comment:
What if this is inside a `async` or `fakeAsync` tests? Will this overwrite the corresponding zone?

Reply:
Very good point. I've played with it and updated the design doc. Also I updated the PR to work with both `fakeAsync` and `async`.